desc/geometry/core.py

```diff
@@ -142,3 +156,119 @@ def __repr__(self):
             + str(hex(id(self)))
             + " (name={}, grid={})".format(self.name, self.grid)
         )
+
+
+def compute_orientation(Rm1, Rp1, Zm1, Zp1):
+    """Compute sign of jacobian based on signs of m= +/-1 modes
+
+    Parameters
+    ----------
+    Rm1 : float
+        R(m=-1)
+    Rp1 : float
+        R(m=+1)
+    Zm1 : float
+        Z(m=-1)
+    Zp1 : float
+        Z(m=+1)
+
+    Returns
+    -------
+    orientation : float
+        +1 for right handed coordinate system (theta increasing CW),
+        -1 for left handed coordinates (theta increasing CCW),
+        or 0 for a singular coordinate system (no volume)
+    """
+
+    _orientation_mat = np.array(
```

Comment:
There has to be a better way to do this than a giant lookup table

Reply:
This assume an expansion like: $R(\rho,\theta) = R^s \rho \sin\theta + R^c \rho \cos\theta$
So I think we can choose to evaluate the Jacobian at an arbitrary point, such as $(\rho=1, \theta=0)$: 
$R(1,0) = R^c$, $R_\rho(1,0) = R^c$, $R_\theta(1,0) = R^s$, and similarly for $Z(1,0)$. 
Then the Jacobian at that point is: $\sqrt{g} = R(R_\theta Z_\rho - R_\rho Z_\theta) = R^c (R^s Z^c - R^c Z^s)$
So I think we could replace the lookup table with this simple line of code:
```
out = np.sign(Rcos * (Rsin * Zcos - Rcos * Zsin))
```
I'll try this and see if it gives the same result as the table.

Reply:
I tried a few different versions of that, the issue is that there's not necessarily any guarantee that theta=0 is on the outboard side, so we don't know which direction d/drho should be in.

Reply:
I think there are some errors in this lookup table.  After some testing with @dpanici, we have identified the following 8 rows as wrong:
```
[-1.0, -1.0, -1.0, -1.0, -1.0]
[-1.0, -1.0, 1.0, 1.0, -1.0]
[-1.0, 1.0, -1.0, 1.0, 1.0]
[-1.0, 1.0, 1.0, -1.0, 1.0]
[1.0, -1.0, -1.0, 1.0, -1.0]
[1.0, -1.0, 1.0, -1.0, -1.0]
[1.0, 1.0, -1.0, -1.0, 1.0]
[1.0, 1.0, 1.0, 1.0, 1.0]
```

We could simply switch the signs in the last column of these rows, and then I think everything would work as intended. 
 Better yet, I suggest the following simplified function that seems to be correct based on our testing: 
```
def _compute_orientation(R0, Rsin, Rcos, Zsin, Zcos):
    """Compute sign of Jacobian determinant of flux coordinate system.

    Parameters
    ----------
    R0 : float
        R(m=0)
    Rsin : float
        R(m=-1)
    Rcos : float
        R(m=+1)
    Zsin : float
        Z(m=-1)
    Zcos : float
        Z(m=+1)

    Returns
    -------
    orientation : float
        +1 for right handed coordinate system (theta increasing CW),
        -1 for left handed coordinates (theta increasing CCW),
        or 0 for a singular coordinate system (no volume)

    """
    out = int(np.sign((R0 + Rcos) * (Rsin * Zcos - Rcos * Zsin)))
    assert (out == -1) or (out == 0) or (out == 1)
    return out
```

Reply:
OK, if that works I like it. But we should check that it works for strongly shaped cases where the centroid (`R0`) is outside of the surface